x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/profile/SuggestProfilesRequest.java

```diff
@@ -66,6 +83,96 @@ public ActionRequestValidationException validate() {
+                        for (Object v : listValue) {
+                            if (v instanceof final String stringValue) {
+                                values.add(stringValue);
+                            } else {
+                                throw new IllegalArgumentException("[labels] hint supports either string value or list of strings");
+                            }
+                        }
+                        labels.put(entry.getKey(), List.copyOf(values));
+                    } else {
+                        throw new IllegalArgumentException("[labels] hint supports either string or list of strings as its value");
+                    }
+                }
+                this.labels = Map.copyOf(labels);
+            } else {
+                this.labels = null;
+            }
+        }
+
+        public Hint(StreamInput in) throws IOException {
+            this.uids = in.readStringList();
+            this.labels = in.readMapOfLists(StreamInput::readString, StreamInput::readString);
+        }
+
+        public List<String> getUids() {
+            return uids;
+        }
+
+        public Tuple<String, List<String>> getSingleLabel() {
+            if (labels == null) {
+                return null;
+            }
+            assert labels.size() == 1 : "labels hint support exactly one key";
+            final String labelKey = labels.keySet().iterator().next();
+            final List<String> labelValues = labels.get(labelKey);
+            assert false == labelValues.isEmpty() : "label values cannot be empty";
+            return new Tuple<>(labelKey, labelValues);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeStringCollection(uids);
+            out.writeMapOfLists(labels, StreamOutput::writeString, StreamOutput::writeString);
+        }
+
+        private ActionRequestValidationException validate(ActionRequestValidationException validationException) {
+            if (uids == null && labels == null) {
+                return addValidationError("[hint] parameter cannot be empty", validationException);
+            }
+
+            if (uids != null && uids.isEmpty()) {
+                validationException = addValidationError("[uids] hint cannot be empty", validationException);
+            }
+
+            if (labels != null) {
+                if (labels.size() != 1) {
+                    return addValidationError(
+                        "[labels] hint supports a single key, got [" + Strings.collectionToCommaDelimitedString(labels.keySet()) + "]",
+                        validationException
+                    );
+                }
+
+                final List<String> values = labels.values().iterator().next();
+                if (values.isEmpty()) {
+                    return addValidationError("[labels] hint cannot be empty", validationException);
+                }
```

Comment:
Maybe we should also validate that `labels.keys().iterator().next()` (and values too) does not contain `*`?

Reply:
We do want to support `*` in the value because Kibana needs it to indicate access to all spaces (existing and future). 

We can prevent `*` in the key since it is [not supported by flattened field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/flattened.html#supported-operations) anyway. Thanks